src/main/java/com/brightpath/learnify/domain/auth/PermissionAccessService.java

```diff
@@ -109,14 +113,63 @@ public void removeUserWithAccessToResource(UUID resourceId, ResourceType resourc
         PermissionsAccessEntity permissionsAccessEntity = permissionAccessRepository.findById(permissionAccessId(resourceId, resourceType))
                 .orElseThrow(() -> new ResourceNotFoundException(resourceType));
 
-        if(permissionsAccessEntity.getOwnerId().equals(userId)) {
+        if (permissionsAccessEntity.getOwnerId().equals(userId)) {
             throw new IllegalArgumentException("Cannot remove owner access");
         }
 
         permissionsAccessEntity.removePermissionForUser(userId);
         permissionAccessRepository.save(permissionsAccessEntity);
     }
 
+    public void checkUserPermissionToEditResource(UUID resourceId, ResourceType resourceType) {
+        String userId = userIdentityService.getCurrentUserId();
+        boolean hasAccessToEditNote = hasUserAccessToResource(userId, resourceId, resourceType, READ_WRITE);
+        if (!hasAccessToEditNote) {
+            throw new UserNotAuthorizedToEditException();
+        }
+    }
+
+    public void checkUserPermissionToViewResource(UUID resourceId, ResourceType resourceType) {
+        String userId = userIdentityService.getCurrentUserId();
+        boolean hasAccessToEditNote = hasUserAccessToResource(userId, resourceId, resourceType, READ_ONLY);
+        if (!hasAccessToEditNote) {
+            throw new UserNotAuthorizedToGetException();
+        }
+    }
+
+    public void addPermissionToResourceForUser(UUID resourceId, ResourceType resourceType, String userId, ResourceAccessEnum requestedAccess) {
+        if (userAnyHasPermissionToResource(resourceId, resourceType, userId, Optional.empty())) {
+            throw new UserAccessIsAlreadyGrantedException();
+        }
+        //todo save access to db
+    }
+
+    public void editPermissionToResourceForUser(UUID resourceId, ResourceType resourceType, String userId, ResourceAccessEnum requestedAccess) {
+        if (userAnyHasPermissionToResource(resourceId, resourceType, userId, Optional.of(requestedAccess))) {
+            throw new UserAccessIsAlreadyGrantedException();
+        }
+        //todo save access to db
+    }
+
+    public void deletePermissionToResourceForUser(UUID resourceId, ResourceType resourceType, String userId, ResourceAccessEnum requestedAccess) {
```

Comment:
I don't think we need to specify the access level when deleting a user's permission